src/main/java/org/gridsuite/study/server/service/ShortCircuitService.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    private UUID getReportUuid(UUID nodeUuid) {
+        return networkModificationTreeService.getReportUuid(nodeUuid);
+    }
+
+    public static ShortCircuitParametersEntity toEntity(ShortCircuitParameters parameters) {
+        Objects.requireNonNull(parameters);
+        return new ShortCircuitParametersEntity(parameters.isWithLimitViolations(),
+                parameters.isWithVoltageMap(),
+                parameters.isWithFeederResult(),
+                parameters.getStudyType(),
+                parameters.getMinVoltageDropProportionalThreshold());
+    }
+
+    public static ShortCircuitParameters fromEntity(ShortCircuitParametersEntity entity) {
+        Objects.requireNonNull(entity);
+        return newShortCircuitParameters(entity.getStudyType(), entity.getMinVoltageDropProportionalThreshold(), entity.isWithFeederResult(), entity.isWithLimitViolations(), entity.isWithVoltageMap());
+    }
+
+    public static ShortCircuitParameters copy(ShortCircuitParameters shortCircuitParameters) {
+        return newShortCircuitParameters(shortCircuitParameters.getStudyType(), shortCircuitParameters.getMinVoltageDropProportionalThreshold(), shortCircuitParameters.isWithFeederResult(), shortCircuitParameters.isWithLimitViolations(), shortCircuitParameters.isWithVoltageMap());
+    }
+
+    private static ShortCircuitParameters newShortCircuitParameters(StudyType studyType, double minVoltageDropProportionalThreshold, boolean withFeederResult, boolean withLimitViolations, boolean withVoltageMap) {
+        ShortCircuitParameters shortCircuitParametersCopy = new ShortCircuitParameters();
+        shortCircuitParametersCopy.setStudyType(studyType);
+        shortCircuitParametersCopy.setMinVoltageDropProportionalThreshold(minVoltageDropProportionalThreshold);
+        shortCircuitParametersCopy.setWithFeederResult(withFeederResult);
+        shortCircuitParametersCopy.setWithLimitViolations(withLimitViolations);
+        shortCircuitParametersCopy.setWithVoltageMap(withVoltageMap);
+        return shortCircuitParametersCopy;
+    }
+
+    public static ShortCircuitParameters getDefaultShortCircuitParameters() {
+        return newShortCircuitParameters(StudyType.TRANSIENT, 20, true, true, false);
+    }
+
+    public void setShortCircuitServerBaseUri(String shortCircuitServerBaseUri) {
+        this.shortCircuitServerBaseUri = shortCircuitServerBaseUri;
+    }
+
+    public void deleteShortCircuitAnalysisResult(UUID uuid) {
+        String path = UriComponentsBuilder.fromPath(DELIMITER + SHORT_CIRCUIT_API_VERSION + "/results/{resultUuid}")
+                .buildAndExpand(uuid)
+                .toUriString();
+
+        restTemplate.delete(shortCircuitServerBaseUri + path);
+    }
+
+    public void assertShortCircuitAnalysisNotRunning(UUID nodeUuid) {
+        String sas = getShortCircuitAnalysisStatus(nodeUuid);
+        if (SensitivityAnalysisStatus.RUNNING.name().equals(sas)) {
```

Comment:
ShortCircuitStatus instead of SensitivityAnalysisStatus (import of SensitivityAnalysisStatus also has to be removed)

Reply:
done